app/org/sagebionetworks/bridge/services/ActivityEventService.java

```diff
@@ -157,4 +169,25 @@ public void deleteActivityEvents(String healthCode) {
         activityEventDao.deleteActivityEvents(healthCode);
     }
 
+    private void createAutomaticCustomEvents(Study study, String healthCode, ActivityEvent event) {
+        for (Map.Entry<String, String> oneAutomaticEvent : study.getAutomaticCustomEvents().entrySet()) {
+            String automaticEventKey = oneAutomaticEvent.getKey(); // new event key
+            // [originEventId:]Period, if originEventId is missing, defaults to "enrollment"
+            Tuple<String> autoEventSpec = parseAutoEventValue(oneAutomaticEvent.getValue());
+            
+            if (event.getEventId().startsWith(autoEventSpec.getLeft())) {
+                Period automaticEventDelay = Period.parse(autoEventSpec.getRight());
+                DateTime automaticEventTime = new DateTime(event.getTimestamp()).plus(automaticEventDelay);
+                publishCustomEvent(study, healthCode, automaticEventKey, automaticEventTime);                
+            }
+        }        
+    }
+    
+    private Tuple<String> parseAutoEventValue(String automaticEventValue) {
+        // though Period serialization does not contain semicolons, limit the split to the first occurrence
+        String[] valueElements = automaticEventValue.split(":", 2); 
+        return (valueElements.length == 1) ?
+            new Tuple<>(ActivityEventObjectType.ENROLLMENT.name().toLowerCase(), valueElements[0]) :
+            new Tuple<>(valueElements[0], valueElements[1]);
```

Comment:
Do we need to special case enrollment events? I thought we had decided to just move these wholesale to activities_retrieved. Enrollment only happens on consent, which means this is a less reliable way to create automatic custom events.

Reply:
This was to ensure that the current configuration in mPower 2.0 works as currently specified (no event in the string). You are right that this is the only use of it so we could probably assume it will be fixed (I updated the editor to handle this origin event specification yesterday, it's in staging).

However I should validate that the automatic custom events when you save a study (to ensure they are the new format), and I'm not doing that. I will add it.

Reply:
Also realizing I need to parse this so that "custom:myEvent:P3W" works correctly (custom events will have two colons).